lock/acquire.go

```diff
@@ -64,28 +88,46 @@ func (locker *EtcdLocker) WaitAcquire(key string, ttl int) (Lock, error) {
 }
 
 func (locker *EtcdLocker) acquire(key string, ttl int, wait bool) (Lock, error) {
+	// A Session is a GRPC connection to ETCD API v3, the connection should be
+	// closed to release resources.
 	session, err := concurrency.NewSession(locker.client, concurrency.WithTTL(ttl))
 	if err != nil {
 		return nil, err
 	}
+
 	key = addPrefix(key)
 	mutex := concurrency.NewMutex(session, key)
-
-	ctx := context.Background()
-	if !wait {
-		var cancel context.CancelFunc
-		ctx, cancel = context.WithTimeout(ctx, locker.trylockTimeout)
-		defer cancel()
+	timeout := time.NewTimer(locker.maxTryLockTimeout)
+
+	for {
+		// If we've wait more than the maxTryLockTimeout, we stop waiting and
+		// consider the lock already taken
+		select {
+		case <-timeout.C:
+			session.Close()
+			return nil, &ErrAlreadyLocked{}
```

Comment:
Do we really want to consider the lock already taken in such case? Can you explain a bit why?

Reply:
Thanks for your remark, it's completely accurate, I've changed the error management